modules/theme-tailwind/organisms/ContactInfoSelection.vue

```diff
@@ -39,44 +59,134 @@
+      </div>
     </Form>
+
+    <CheckoutLoginModal
+      v-if="showLogin"
+      :email="contactEmail"
+      @close="
+        () => {
+          showLogin = false
+          offerLogin = false
+        }
+      "
+    />
+    <CheckoutRegisterModal
+      v-if="showRegistration"
+      :email="contactEmail"
+      @close="
+        () => {
+          showRegistration = false
+          offerRegistration = false
+        }
+      "
+    />
   </div>
 </template>
 
 <script setup lang="ts">
 import SfxShippingMethod from '#ioc/components/SfxShippingMethod'
 import Form from '#ioc/atoms/Form'
 import useCheckout from '#ioc/composables/useCheckout'
+import useCustomer from '#ioc/composables/useCustomer'
 import useI18n from '#ioc/composables/useI18n'
+import useEmailAvailable from '#ioc/services/useEmailAvailable'
+import Button from '#ioc/atoms/Button'
 import FormInput from '#ioc/molecules/FormInput'
+import CheckoutLoginModal from '#ioc/organisms/CheckoutLoginModal'
+import CheckoutRegisterModal from '#ioc/organisms/CheckoutRegisterModal'
+import CheckoutCustomerAddress from '#ioc/molecules/CheckoutCustomerAddress'
+import useGetCustomerAddresses from '#ioc/services/useGetCustomerAddresses'
 import debounce from '#ioc/utils/debounce'
-import { ref, watch } from 'vue'
+import { ref, watch, onMounted } from 'vue'
 
 const props = defineProps({
   isOpen: Boolean,
 })
 
 const emit = defineEmits(['select', 'confirm'])
-
+const customer = useCustomer()
 const { t } = useI18n()
 const checkout = useCheckout()
+const isEmailAvailable = useEmailAvailable()
+
+const getCustomerAddresses = useGetCustomerAddresses()
+
+const offerLogin = ref(false)
+const offerRegistration = ref(false)
+const showLogin = ref(false)
+const showRegistration = ref(false)
+const contactEmail = ref()
+const selectedAddress = ref(null)
+const customerAddresses = ref()
+
+const onSelectCustomerAddress = (customerAddress: any) => {
+  const shippingAddress = {
+    ...customerAddress,
+    telephone: customerAddress?.phoneNumber,
+  }
+  selectedAddress.value = shippingAddress
+  onInput(shippingAddress)
+}
+
+const onCustomAddress = () => {
+  selectedAddress.value = null
+}
+
+const checkEmail = async (email: string) => {
+  if (customer.isLoggedIn || !email) {
+    offerLogin.value = false
+    offerRegistration.value = false
+    return
+  }
+
+  const { emailAvailable } = await isEmailAvailable(email)
+
+  if (emailAvailable) {
+    offerLogin.value = false
+    offerRegistration.value = true
+  } else {
+    offerLogin.value = true
+    offerRegistration.value = false
+  }
+}
 
 const form = ref<any>(null)
 
 const onInput = debounce(async (data: any) => {
-  if (!form.value?.isValid) return emit('select')
+  if (!form.value?.isValid && !selectedAddress.value) return emit('select')
+  const address = selectedAddress.value ? selectedAddress.value : data
+
+  contactEmail.value = form.value?.getData()?.email ?? {}
```

Comment:
You already have the form data as part of this function input arguments. No need to get it from form again.